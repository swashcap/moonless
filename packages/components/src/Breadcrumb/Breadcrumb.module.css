/** Breadcrumb: List */

.BreadcrumbList {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

/** Breadcrumb: List: Item */

.BreadcrumbListItem::after {
  color: var(--theme-color-text-secondary);
  content: '/';
  display: inline-block;
}

.BreadcrumbListItem:last-child::after {
  content: '';
  display: none;
}

/** Breadcrumb: Link */

.BreadcrumbLink {
  display: inline-block;
  padding: var(--space-50);
}

.BreadcrumbLink[aria-current] {
  font-weight: 700;
}
