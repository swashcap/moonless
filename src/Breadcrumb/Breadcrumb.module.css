/** Breadcrumb: List */

.BreadcrumbList {
  display: flex;
  list-style: none;
  margin: 0;
  padding: 0;
}

/** Breadcrumb: List: Item */

.BreadcrumbListItem::after {
  content: '/';
  display: inline-block;
}

.BreadcrumbListItem:last-child::after {
  content: '';
  display: none;
}

/** Breadcrumb: Link */

.BreadcrumbLink {
  display: inline-block;
  padding: var(--spacing-100);
}

.BreadcrumbLink[aria-current] {
  font-weight: 700;
}
