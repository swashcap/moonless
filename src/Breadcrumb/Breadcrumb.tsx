import React from 'react';

import styles from './Breadcrumb.module.css';

export interface BreadcrumbProps
  extends React.DetailedHTMLProps<
    React.HTMLAttributes<HTMLDivElement>,
    HTMLDivElement
  > {
  items: ReadonlyArray<
    Readonly<{
      current?: boolean;
      href: string;
      name: string;
    }>
  >;
}

export const Breadcrumb: React.FC<BreadcrumbProps> = ({
  children,
  items,
  ...rest
}) => (
  <nav {...rest}>
    <ol
      className={styles.BreadcrumbList}
      itemScope
      itemType="http://schema.org/BreadcrumbList"
    >
      {items.map(({ current, href, name }, index) => (
        <li
          className={styles.BreadcrumbListItem}
          itemProp="itemListElement"
          itemScope
          itemType="http://schema.org/ListItem"
          key={href}
        >
          <a
            aria-current={current ? 'page' : undefined}
            className={styles.BreadcrumbLink}
            itemProp="item"
            href={href}
          >
            <span itemProp="name">{name}</span>
          </a>
          <meta itemProp="position" content={(index + 1).toString()} />
        </li>
      ))}
    </ol>
  </nav>
);
