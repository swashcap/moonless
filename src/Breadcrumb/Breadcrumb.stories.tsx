import React from 'react';

import { Breadcrumb } from './Breadcrumb';

export default {
  component: Breadcrumb,
  title: 'Breadcrumb',
};

export const Default = () => (
  <Breadcrumb
    items={[
      {
        href: '/',
        name: 'Home',
      },
      {
        href: '/page-2/',
        name: 'Page 2',
      },
      {
        current: true,
        href: '/page-2/page-3/',
        name: 'Page 3',
      },
    ]}
  />
);
