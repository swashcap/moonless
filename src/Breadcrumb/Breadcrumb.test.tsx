import React from 'react';
import { render } from '@testing-library/react';

import { Breadcrumb } from './Breadcrumb';

describe('Breadcrumb', () => {
  test('It should render.', () => {
    const { container } = render(
      <Breadcrumb
        items={[
          {
            href: '/',
            name: 'Home',
          },
        ]}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
