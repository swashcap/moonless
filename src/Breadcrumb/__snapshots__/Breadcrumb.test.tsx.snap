// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb It should render. 1`] = `
<div>
  <nav
    class="Breadcrumb"
  >
    <ol
      class="BreadcrumbList"
      itemscope=""
      itemtype="http://schema.org/BreadcrumbList"
    >
      <li
        class="BreadcrumbListItem"
        itemprop="itemListElement"
        itemscope=""
        itemtype="http://schema.org/ListItem"
      >
        <a
          class="BreadcrumbLink"
          href="/"
          itemprop="item"
        >
          <span
            itemprop="name"
          >
            Home
          </span>
        </a>
        <meta
          content="1"
          itemprop="position"
        />
      </li>
    </ol>
  </nav>
</div>
`;
